ogr/services/forgejo/issue.py

```diff
@@ -1,18 +1,101 @@
+import pyforgejo.types.issue as _issue
+from pyforgejo import NotFoundError
 
 from ogr.abstract import Issue, IssueStatus
+from ogr.exceptions import IssueTrackerDisabled, OperationNotSupported
 from ogr.services import forgejo
 from ogr.services.base import BaseIssue
+from ogr.services.forgejo.utils import paginate
 
 
 class ForgejoIssue(BaseIssue):
-    def __init__(self, raw_issue: PyforgejoIssue, project: "forgejo.ForgejoProject"):
+    project: "forgejo.ForgejoProject"
+
+    def __init__(self, raw_issue: _issue, project: "forgejo.ForgejoProject"):
         super().__init__(raw_issue, project)
+        self._raw_issue = raw_issue
+
+    @property
+    def _index(self):
+        return self._raw_issue.number
+
+    @property
+    def api(self):
+        """Returns the issue API client from pyforgejo."""
+        return self.project.service.api.issue
+
+    def partial_api(self, method, /, *args, **kwargs):
+        """Returns a partial API call for ForgejoIssue.
+
+        Injects owner, repo, and index parameters for the calls to issue API endpoints.
+
+        Args:
+            method: Specific method on the Pyforgejo API that is to be wrapped.
+            *args: Positional arguments that get injected into every call.
+            **kwargs: Keyword-arguments that get injected into every call.
+
+        Returns:
+            Callable with pre-injected parameters.
+        """
+        params = {"owner": self.project.namespace, "repo": self.project.repo}
+
+        # Include the issue index only for methods that need it
+        if "get_issue" in str(method) or "edit_issue" in str(method):
+            params["index"] = self._index
+
+        return partial(method, *args, **kwargs, **params)
+
+    def __update_info(self) -> None:
+        """Refresh the local issue object with the latest data from the server."""
+        self._raw_issue = self.partial_api(self.api.get_issue)()
+
+    @property
+    def title(self) -> str:
+        return self._raw_issue.title
+
+    @title.setter
+    def title(self, new_title: str) -> None:
+        self.partial_api(self.api.edit_issue)(title=new_title)
+        self.__update_info()
+
+    @property
+    def id(self) -> int:
+        return self._raw_issue.number
+
+    @property
+    def url(self) -> str:
+        return self._raw_issue.url
+
+    @property
+    def description(self) -> str:
+        return self._raw_issue.body
+
+    @description.setter
+    def description(self, text: str):
+        self.partial_api(self.api.edit_issue)(body=text)
+        self.__update_info()
+
+    @property
+    def author(self) -> str:
+        return self._raw_issue.user.login
+
+    @property
+    def created(self) -> datetime:
+        return self._raw_issue.created_at
+
+    @property
+    def status(self) -> IssueStatus:
+        return IssueStatus[self._raw_issue.state]
+
+    @property
+    def assignees(self) -> list:
+        return getattr(self._raw_issue, "assignees", []) or []
```

Comment:
```suggestion
        return self._raw_issue.assignees or []
```

I just checked the pyforgejo, this is way too overcomplicated